api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,381 @@
+	// BmcAddress holds the URL for accessing the controller on the network.
+	// +required
+	BmcAddress string `json:"bmcAddress,omitempty"`
+
+	// BmcCredentialsName is the name of the secret containing the BMC credentials (requires keys "username" and "password").
+	// +required
+	BmcCredentialsName BmcCredentialsName `json:"bmcCredentialsName,omitempty"`
+
+	// Which MAC address will PXE boot? This is optional for some
+	// types, but required for libvirt VMs driven by vbmc.
+	// +kubebuilder:validation:Pattern=`[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}`
+	// +required
+	BootMACAddress string `json:"bootMACAddress,omitempty"`
+
+	// When set to disabled, automated cleaning will be avoided during provisioning and deprovisioning.
+	// Set the value to metadata to enable the removal of the disk’s partitioning table only, without fully wiping the disk. The default value is disabled.
+	// +optional
+	// +kubebuilder:default:=disabled
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode bmh_v1alpha1.AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
+
+	// RootDeviceHints specifies the device for deployment.
+	// Identifiers that are stable across reboots are recommended, for example, wwn: <disk_wwn> or deviceName: /dev/disk/by-path/<device_path>
+	// +optional
+	RootDeviceHints bmh_v1alpha1.RootDeviceHints `json:"rootDeviceHints,omitempty"`
+
+	// NodeNetwork is a set of configurations pertaining to the network settings for the node.
+	// +optional
+	NodeNetwork aiv1beta1.NMStateConfigSpec `json:"nodeNetwork,omitempty"`
+
+	// NodeLabels allows the specification of custom roles for your nodes in your managed clusters.
+	// These are additional roles are not used by any OpenShift Container Platform components, only by the user.
+	// When you add a custom role, it can be associated with a custom machine config pool that references a specific configuration for that role.
+	// Adding custom labels or roles during installation makes the deployment process more effective and prevents the need for additional reboots
+	// after the installation is complete.
+	// +optional
+	NodeLabels map[string]string `json:"nodeLabels,omitempty"`
+
+	// Hostname is the desired hostname for the host
+	// +required
+	HostName string `json:"hostName,omitempty"`
+
+	// Provide guidance about how to choose the device for the image being provisioned.
+	// +kubebuilder:default:=UEFI
+	BootMode bmh_v1alpha1.BootMode `json:"bootMode,omitempty"`
+
+	// Json formatted string containing the user overrides for the host's coreos installer args
+	// +optional
+	InstallerArgs string `json:"installerArgs,omitempty"`
+
+	// Json formatted string containing the user overrides for the host's ignition config
+	// IgnitionConfigOverride enables the assignment of partitions for persistent storage.
+	// Adjust disk ID and size to the specific hardware.
+	// +optional
+	IgnitionConfigOverride string `json:"ignitionConfigOverride,omitempty"`
+
+	// +kubebuilder:default:=master
```

Comment:
@imiller0 @carbonin @eranco74 should we add the following validation:
```golang
// +kubebuilder:validation:Enum=ControlPlane;Worker
```
This is a great time to move away from the old SiteConfig roles of `master` and `worker`

Reply:
and change default to `ControlPlane`?

Reply:
I'd say yes. Would this translate to master under the hood (not sure what AI uses/requires)

Reply:
@carbonin / @eranco74 can you folks help answer this

Reply:
Indeed `master`/`worker` is what AI requires so you can change this and translate or leave it as is.

Reply:
Adding the validation:
```golang
// +kubebuilder:validation:Enum=master;worker
```

Reply:
I believe that it can also be `auto-assign` (which is the default).

Reply:
Thanks @eranco74. Now that we are automatically computing the number of control-plane and worker agents in order to feed to the ACI template, allowing `auto-assign` would prevent us from calculating these values. Hence just allowing `master` and `worker` values.